### DISCUSSION TITLE
Custom 'Enter' key behavior

### DISCUSSION THREAD

**Discussion by duong-ld** [adobe/react-spectrum#6826] [Q&A] [answered]
Hi guys! 
I'm working on creating a multi-select menu with react-area. The menu requires a support keyboard ('Enter' to open and close the menu and 'Space' to select an item). It is almost like [this example](https://react-spectrum.adobe.com/react-aria/Menu.html#multiple)
As you can see in the example, the default behavior is almost like my requirement. But when I close the menu using the 'Enter' key, the menu closes and also selects (or unselects) the current item (as in the video below). How do I customize the 'Enter' key behavior?
Thank you, and sorry for my bad English!

https://github.com/user-attachments/assets/2e512908-2717-422f-8043-5dca775cb090

**Comment by snowystinger** [comment, marked as answer]
It's not customizable at this time and it's working as intended. We are following the expected behavior as defined by the ARIA patterns, seen here https://www.w3.org/WAI/ARIA/apg/patterns/menubar/

> Enter:
>  - When focus is on a menuitem that has a submenu, opens the submenu and places focus on its first item.
>  - Otherwise, activates the item and closes the menu.

> Space:
>  - (Optional): When focus is on a menuitemcheckbox, changes the state without closing the menu.
>  - (Optional): When focus is on a menuitemradio that is not checked, without closing the menu, checks the focused menuitemradio and unchecks any other checked menuitemradio element in the same group.
>  - (Optional): When focus is on a menuitem that has a submenu, opens the submenu and places focus on its first item.
>  - (Optional): When focus is on a menuitem that does not have a submenu, activates the menuitem and closes the menu.

**Reply by duong-ld** [reply]
That's fair enough; I appreciate your response.
